### PR TITLE
Implementation of the LOST triangulation algorithm

### PR DESCRIPTION
**Pull request by akshay-krishnan**
In collaboration with Sebastien Henry and Prof. John Christian, we implement the LOST triangulation algorithm  https://arxiv.org/pdf/2205.12197.pdf. 
The triangulatePoint3 method accepts a boolean that initializes the point using the LOST method instead of DLT. 

This PR also adds an example that runs DLT, LOST and DLT+nonlinear optimization. It shows that the covariance of LOST is similar to that of DLT + optimization, while being much faster.

### CODE REVIEW

**Comment by akshay-krishnan** [comment]
Its possible my design choice may not be the best. The goal was to have the LOST implementation share the same triangulatePoint3 interface. 
I could not make it more similar than this. LOST required the poses, while DLT works on projection matrices. 
@dellaert let me know if you prefer a different design.

File: gtsam/geometry/tests/testTriangulation.cpp [hunk 1]
```diff
@@ -96,6 +96,70 @@ TEST(triangulation, twoPoses) {
...
+          /*optimize=*/false, measurementNoise,
+          /*use_lost_triangulation=*/true);
+  EXPECT(assert_equal(Point3(4.995, 0.499167, 1.19814), *actual2, 1e-4));
+}
+
+TEST(triangulation, twoCamerasLOSTvsDLT) {
+  // LOST has been shown to preform better when the point is much closer to one
+  // camera compared to another. This unit test checks this configuration.
+  Cal3_S2 identity_K;
+  Pose3 pose_1;
```
Comment (dellaert) [inline gtsam/geometry/tests/testTriangulation.cpp:132]: pose1, pose2, identityK

File: gtsam/geometry/triangulation.h [hunk 2]
```diff
@@ -302,8 +316,8 @@ template <class CAMERA>
 typename CAMERA::MeasurementVector undistortMeasurements(
     const CameraSet<CAMERA>& cameras,
     const typename CAMERA::MeasurementVector& measurements) {
-  const size_t num_meas = cameras.size();
-  assert(num_meas == measurements.size());
+  const size_t num_meas = measurements.size();
```
Comment (dellaert) [inline gtsam/geometry/triangulation.h:319]: rename to `nrMeasurements` and `nrCameras`

File: gtsam/geometry/triangulation.h [hunk 3]
```diff
@@ -341,41 +414,56 @@ inline SphericalCamera::MeasurementVector undistortMeasurements(
...
-    boost::shared_ptr<CALIBRATION> sharedCal,
-    const Point2Vector& measurements, double rank_tol = 1e-9,
-    bool optimize = false,
-    const SharedNoiseModel& model = nullptr) {
-
+                         boost::shared_ptr<CALIBRATION> sharedCal,
+                         const Point2Vector& measurements,
+                         double rank_tol = 1e-9, bool optimize = false,
+                         const SharedNoiseModel& model = nullptr,
+                         const bool use_lost_triangulation = false) {
```
Comment (dellaert) [inline gtsam/geometry/triangulation.h:427]: In GTSAM we use camelCase. `useLOST` seems good.

File: examples/TriangulationLOSTExample.cpp [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    dlt_opt_errors.row(i) = *estimate_dlt_opt - gt_point;
+  }
+  PrintCovarianceStats(lost_errors, "LOST");
+  PrintCovarianceStats(dlt_errors, "DLT");
+  PrintCovarianceStats(dlt_opt_errors, "DLT_OPT");
+
+  PrintDuration(lost_duration, num_trials, "LOST");
+  PrintDuration(dlt_duration, num_trials, "DLT");
+  PrintDuration(dlt_opt_duration, num_trials, "DLT_OPT");
+}
```
Comment (dellaert) [inline examples/TriangulationLOSTExample.cpp:161]: Add newline

**Review by dellaert — APPROVED** [review]
Awesome! some naming nits - we use camelCase, but after that merge at will.

**Pull request merged**